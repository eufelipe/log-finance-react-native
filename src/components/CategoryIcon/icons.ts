export interface IconProp {
  [key: string]: {
    name: string;
    type: string;
  };
}

export const icons: IconProp = {
  others: {
    name: 'list',
    type: 'MaterialIcons',
  },
  food: {
    name: 'settings',
    type: 'MaterialIcons',
  },
};

export default icons;
