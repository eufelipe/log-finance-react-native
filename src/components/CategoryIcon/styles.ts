import styled from 'styled-components/native';

import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Colors from 'styles/colors';

export const IconItem = styled(MaterialIcons).attrs({
  size: 24,
  color: Colors.gray,
})``;
