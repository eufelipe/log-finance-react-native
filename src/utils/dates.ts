import {
  format as formatPattern,
  isValid,
  parseISO as parseISOFNS,
} from 'date-fns';

import locale from 'date-fns/locale/pt-BR';

export const DATE_FORMAT_ISO = 'yyyy-MM-dd';
export const DATE_FORMAT_BR = 'dd/MM/yyyy';
export const DATE_FORMAT_FRIENDLY = " dd/MM 'às' HH:mm";
export const DATE_FORMAT_FRIENDLY_TIME = "'às' HH:mm";

export function parseStringToDate(value: string): Date {
  const parsedDate = parseISOFNS(value);
  if (!isValid(parsedDate)) {
    throw Error('canot parse date');
  }
  return parsedDate;
}
export function parseDateToString({
  value,
  format = DATE_FORMAT_BR,
}: {
  value: Date;
  format?: string;
}): string {
  return formatPattern(value, format, {locale});
}

export const getDate = (date: Date): string => {
  return formatPattern(date, DATE_FORMAT_ISO, {locale});
};

export const getDateToday = (): string => {
  const today = new Date();
  return formatPattern(today, DATE_FORMAT_ISO, {locale});
};

export const getDateAndHourFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY, {locale});
};

export const getTimeFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY_TIME, {locale});
};
