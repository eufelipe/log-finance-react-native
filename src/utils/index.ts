export {default as number} from './number';
export {default as strings} from './strings';
