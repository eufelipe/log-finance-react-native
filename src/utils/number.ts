export const sanitizeToFloat = (value: string): number => {
  value = value.replace(/[^0-9.]+/, '');
  if (value.includes('.') && value.includes(',')) {
    value = value.replace(/\./g, '').replace(',', '.');
  } else if (value.includes(',')) {
    value = value.replace(',', '.');
  }
  return parseFloat(value) || 0;
};
