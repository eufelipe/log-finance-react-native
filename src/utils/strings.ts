export const sanitizeString = (value: string): string => {
  return value
    .toLowerCase()
    .replace(/á/g, 'a')
    .replace(/é/g, 'e')
    .replace(/í/g, 'i')
    .replace(/ó/g, 'o')
    .replace(/ú/g, 'u');
};
