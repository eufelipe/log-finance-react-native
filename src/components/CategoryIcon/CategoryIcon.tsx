import React from 'react';
import icons from './icons';

interface Props {
  name?: string;
}

import {IconItem} from './styles';

const CategoryIcon = ({name = 'others'}: Props): JSX.Element => {
  const iconOption = name in icons ? icons[name] : icons.others;

  return <IconItem name={iconOption.name} />;
};

export default CategoryIcon;
