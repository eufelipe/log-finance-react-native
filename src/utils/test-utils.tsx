import React from 'react';
import {render, RenderAPI, RenderOptions} from '@testing-library/react-native';

import {theme} from 'styles/theme';
import {ThemeProvider} from 'styled-components/native';

interface Props {
  children: JSX.Element;
}

const AllTheProviders = ({children}: Props) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions,
): RenderAPI => render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
