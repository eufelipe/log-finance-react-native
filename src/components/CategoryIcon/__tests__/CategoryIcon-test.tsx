/* eslint-disable @typescript-eslint/unbound-method */
import React from 'react';
import {render} from 'utils/test-utils';
import CategoryIcon from '../CategoryIcon';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');

describe('CategoryIcon', () => {
  it('Test match snapshot CategoryIcon', () => {
    // given
    const props = {
      name: 'food',
    };

    // when
    const {toJSON} = render(<CategoryIcon {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  it('Test match snapshot CategoryIcon default value', () => {
    // given
    const props = {};

    // when
    const {toJSON} = render(<CategoryIcon {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
