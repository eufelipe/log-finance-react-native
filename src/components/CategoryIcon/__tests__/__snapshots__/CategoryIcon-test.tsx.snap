// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryIcon Test match snapshot CategoryIcon 1`] = `
<Icon
  color="#7C828E"
  name="settings"
  size={24}
  style={
    Array [
      Object {},
    ]
  }
/>
`;

exports[`CategoryIcon Test match snapshot CategoryIcon default value 1`] = `
<Icon
  color="#7C828E"
  name="list"
  size={24}
  style={
    Array [
      Object {},
    ]
  }
/>
`;
